### PR TITLE
Adds Device class - a view onto a core device with a nicer API

### PR DESCRIPTION
**Pull request by tlambert03**
Same idea as #107, but for Devices

```python
from pymmcore_plus import CMMCorePlus, Device

core = CMMCorePlus()
device = Device(core, 'Camera')
device.isLoaded()
device.load('NotALib', 'DCam')  # useful error
device.load('DemoCamera', 'DCam')
device.initialize()
device.load('DemoCamera', 'DCam')  # no-op w/ useful warning
device.properties  # tuple of DeviceProperty objects
device.description()
device.isBusy()
device.wait()
device.type()
device.schema()  # JSON schema of device properties
```

```python
{
    'title': 'DCam',
    'description': 'Demo camera',
    'type': 'object',
    'properties': {
        'HubID': {'type': 'string', 'readOnly': True, 'default': ''},
        'MaximumExposureMs': {'type': 'number', 'preInit': True},
        'TransposeCorrection': {'type': 'boolean'},
        'TransposeMirrorX': {'type': 'boolean'},
        'TransposeMirrorY': {'type': 'boolean'},
        'TransposeXY': {'type': 'boolean'}
    }
}
```

### CODE REVIEW

File: pymmcore_plus/core/_mmcore_plus.py [hunk 1]
```diff
@@ -351,15 +395,12 @@ def getPropertyObject(
         """Return a DeviceProperty object bound to a device/property on this core."""
         return DeviceProperty(self, device_label, property_name)
 
-    def getDeviceProperties(self, device_label: str) -> Dict[str, Any]:
```
Comment (tlambert03) [inline pymmcore_plus/core/_mmcore_plus.py:354]: note the removal of this method... not really that useful anyway.  Now becomes:

```python
{p.name: p.value for p in core.iterProperties(device_label=..., as_object=True)}
```

File: pymmcore_plus/core/_device.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+                    devices = self._mmc.getAvailableDevices(adapter_name)
+                    if device_name not in devices:
+                        msg += (
+                            f". Device name {device_name!r} not in devices provided by "
+                            f"adapter {adapter_name!r}: {devices}"
+                        )
+            raise RuntimeError(msg)  # sourcery skip
+
+    def unload(self):
+        """Unload device from the core and adjust all configuration data."""
```
Comment (ianhi) [inline pymmcore_plus/core/_device.py:135]: what does "adjust all configuration data" mean?

**Review by ianhi — APPROVED** [review]
This is great - no complaints from me

File: pymmcore_plus/core/_device.py (hunk 2 above)
Comment (tlambert03) [reply pymmcore_plus/core/_device.py:135]: ya got me! :) https://valelab4.ucsf.edu/~MM/doc/MMCore/html/class_c_m_m_core.html#a645c4c10e8be5e705c346336dab28cd5

Comment (ianhi) [reply pymmcore_plus/core/_device.py:135]: I asked: https://github.com/micro-manager/mmCoreAndDevices/issues/174

**Pull request merged**